src/AnalyticsEventReporter.ts

```diff
@@ -60,22 +59,22 @@ export class AnalyticsEventReporter implements AnalyticsEventService {
       const shouldUseBeacon = useBeacon(finalPayload, this.config.forceFetch);
       const requestUrl = setupRequestUrl(this.config.env, this.config.region);
 
-      // If useBeacon returns true, return boolean response of postWithBeacon
+      // If useBeacon returns true, return boolean response of postWithBeacon as string.
       if (shouldUseBeacon) {
-        return await postWithBeacon(requestUrl, finalPayload);
+        return new Promise((resolve, reject) => {
+          if (postWithBeacon(requestUrl, finalPayload)) {
+            resolve('');
+          } else {
+            reject('Failed Beacon Call');
+          }
+        });
       }
 
       /** If useBeacon returns false, use postWithFetch.
-          If result is successful, return result json.
-          If request fails, return errors. */
-      const res = await postWithFetch(requestUrl, finalPayload);
-      if (!res?.ok) {
-        const body: EventAPIResponse = await res?.json();
-        let errorMessage = `Events API responded with ${res?.status}: ${res?.statusText}`;
-        body?.errors?.forEach(e => errorMessage += `\nError: ${e}.`);
-        throw new Error(errorMessage);
-      }
-      const resJson = await res?.json();
-      return resJson;
+      If result is successful, return result json.
+      If request fails, return errors. */
+      return postWithFetch(requestUrl, finalPayload)
+        .then((response) => response)
```

Comment:
Do we not need to do response.text() or response.json?

Reply:
I do not believe so. I just tried using .text() and .json and both result in undefined being returned.